src/unity/lib/unity_sarray.cpp

```diff
@@ -1300,9 +1300,9 @@ std::shared_ptr<unity_sarray_base> unity_sarray::lazy_astype(flex_type_enum dtyp
       flexible_type ret;
       try {
         if (dtype == flex_type_enum::INTEGER) {
-          ret = std::stoll(f.get<flex_string>());
+          f.to<flex_int>();
```

Comment:
to<...> returns. It does not cast inplace. See the flexible_type documentation.

Reply:
Of course - not sure how I missed that. How does this pass unit tests?